Modules/modExtendSaveAsText.ACM

```diff
@@ -89,20 +93,44 @@ End Sub
 Public Sub LoadFromText(objType As asObjectType, ByVal objName As String, strFilename As String, Optional strErr As String)
   Dim strCatchErr As String
   
+  ' Forms, Reports, Queries and Reports are produced and read by Access from UCS2 ( 2 byte ) encoded files
+  ' These are re-encoded to UTF-8 on saving so that all SVN tools work
+  ' Due to the historic files not having been converted we need to check whether these files are in UTF8 or UCS2 format and convert if required
+  
   Select Case objType
     Case asObjectType.acForm
       ' This is needed for when importing the SVN Tool from SVN using the SVN Tool!!!!!????
       If FormIsLoaded(objName) Then
         strErr = "The Form '" & objName & "' is already open. Inserted as " & objName & "-svntmp. Please rename after import"
         objName = objName & "-svntmp"
       End If
-      Application.LoadFromText acForm, objName, strFilename
-    Case asObjectType.acReport, asObjectType.acMacro, asObjectType.acModule
+      If IsUCS2(strFilename) Then
+        Application.LoadFromText objType, objName, strFilename
+      Else

```

Comment:
Explicitly check for `IsUTF8` , and if not show an error?

Reply:
Logic changed. As we are now using a BOM in the UTF8 file we can positively identify UTF8. So test for that, if it's not UTF8 simply use build in import (which copes with both UCS2 and ANSI).